agent/lib/kontena/workers/service_pod_manager.rb

```diff
@@ -15,27 +16,26 @@ class ServicePodManager
     finalizer :finalize
 
     def initialize(autostart = true)
+      @node = nil
       @workers = {}
-      subscribe('agent:node_info', :on_node_info)
-      subscribe('service_pod:update', :on_update_notify)
+
       async.start if autostart
     end
 
     def start
+      observe(Actor[:node_info_worker]) do |node|
+        @node = node
+      end
+
       wait_until!("have node info", interval: 0.1) { self.node }
```

Comment:
This is still kinda clunky... but the reason is that this should eventually look something like

```ruby
observe(Actor[:node_info_worker], Actor[:service_pods_worker], Actor[:docker_containers_worker]) do |node, service_pods, docker_containers|
   ....
end
```